File: projects/m2/024-create-a-bingo-card/valentinaBazzacchi/solution-create-a-bingo-card.js
// Function to create a set of 5 different random numbers from a range of 15
function createRandomNum(n) {
    let setNum = new Set();
    while (true) {
        setNum.add((Math.floor(Math.random() * 15) + n));
        if (setNum.size === 5) break;
    }
    let numList = [];
    for (let num of setNum)
        numList.push(num);
    return numList;
} // end function

// Function to create a random Bingo card and stores it in a object
function createBingoCard() {
    let obj = {
        B: createRandomNum(1),
        I: createRandomNum(16),
        N: createRandomNum(31),
        G: createRandomNum(46),
        O: createRandomNum(61),
    };
    return obj;
} // end function

function displayBingoCard(obj) {
    let string = "";
    for (let key in obj) {
        string += key + "  |  ";
    }
    
    string += "\n----------------------\n"
    
    for (let i = 0; i < 5; i++) {
        for (let key in obj) {
            string += obj[key][i] + " | ";
        }
        
        string += "\n"
    }
    return string;
} // end function

// MAIN PROGRAM
let bingoCard = createBingoCard();
console.log(bingoCard);
alert(displayBingoCard(bingoCard));
console.log(displayBingoCard(bingoCard));

File: projects/m2/026-play-bingo/python/solution_create_a_bingo_card.py
from random import randint

def create_bingo_card():
    bingo_card = {"B":[], "I":[], "N":[], "G":[], "O":[]}
    for i in range(5):
        bingo_card["B"].append(randint(1,15))
        bingo_card["I"].append(randint(16,30))
        bingo_card["N"].append(randint(31,45))
        bingo_card["G"].append(randint(46,60))
        bingo_card["O"].append(randint(61,75))
    return bingo_card

def display_bingo_card(card):
    print("{:^3} {:^3} {:^3} {:^3} {:^3}".format("B", "I", "N", "G", "O"))
    for i in range(5):
        print("{:3} {:3} {:3} {:3} {:3}".format(card["B"][i], card["I"][i], card["N"][i], card["G"][i], card["O"][i]))

if __name__ == "__main__":
    display_bingo_card(create_bingo_card())